src/scripts.js

```diff
@@ -60,8 +66,51 @@ function updateAllRecipeDisplay() {
 
     // TO DO: Put modal event handler in line 63
     recipeCard.onclick = () => {
+      // if (event classname === star-icon) {
+        // do the function to toggleFavorites
+      // }
+
+      // if (event classname === recipe-tags) {
+    //  the function tagsToggleFilter
+      // }
+
+       // if (event classname === recipe-card) {
+    //  the function to open the recipe modal
+      // }
+
       console.log(`hi! ${recipe.name}`)
     }
     recipeSection.appendChild(recipeCard)
   })
 }
+
+function tagsToggleFilter(event) {
+  if(tag === event.target.innerText) {
+    updateAllRecipeDisplay(recipeRepo.allRecipes)
+    event.target.style.backgroundColor = "cornflowerblue";
+  } else {
+    let userTag = getTag(event);
+    console.log(userTag)
+    getFilteredRecipes(userTag);
+  }
+ }
+
+function getTag(event) {
+  if(event.target.className === 'recipe-tag') {
+    tag = event.target.innerText
+    event.target.style.backgroundColor = "hotpink"
+    return tag
+  }
+}
+
+function getFilteredRecipes(tag) {
+    updateAllRecipeDisplay(recipeRepo.filterByTag(tag));
+  }
```

Comment:
I realized at lunch that we could refactor this! Line `107` can replace line `94` and we can get rid of the `getFilteredRecipes` function 🎉